scheme/arm-cca/store_handler.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package arm_cca
+
+import (
+	"fmt"
+
+	"github.com/veraison/ccatoken"
+	"github.com/veraison/services/handler"
+	"github.com/veraison/services/proto"
+	"github.com/veraison/services/scheme/common/arm"
+	realm "github.com/veraison/services/scheme/common/cca/realm"
+)
+
+type StoreHandler struct{}
+
+func (s StoreHandler) GetName() string {
+	return "cca-store-handler"
+}
+
+func (s StoreHandler) GetAttestationScheme() string {
+	return SchemeName
+}
+
+func (s StoreHandler) GetSupportedMediaTypes() []string {
+	return nil
+}
+
+func (s StoreHandler) SynthKeysFromRefValue(
+	tenantID string,
+	refVal *handler.Endorsement,
+) ([]string, error) {
+	switch refVal.SubType {
+	case "sw-component", "platform-config":
+		return arm.SynthKeysForPlatform(SchemeName, tenantID, refVal)
+	case "realm.reference-value":
+		return realm.SynthKeysForCcaRealm(SchemeName, tenantID, refVal)
+	default:
+		return nil, fmt.Errorf("invalid SubType: %s, for Scheme: %s", refVal.SubType, refVal.Scheme)
+	}
+}
+
+func (s StoreHandler) SynthKeysFromTrustAnchor(tenantID string, ta *handler.Endorsement) ([]string, error) {
+	return arm.SynthKeysFromTrustAnchors(SchemeName, tenantID, ta)
+}
+
+func (s StoreHandler) GetTrustAnchorIDs(token *proto.AttestationToken) ([]string, error) {
+	var evidence ccatoken.Evidence
+	err := evidence.FromCBOR(token.Data)
+	if err != nil {
+		return []string{""}, handler.BadEvidence(err)
+	}
+
+	claims := evidence.PlatformClaims
+	if err != nil {
+		return []string{""}, err
+	}
+	taID, err := arm.GetTrustAnchorID(SchemeName, token.TenantId, claims)
+	if err != nil {
+		return []string{""}, err
+	}
+
+	return []string{taID}, nil
+}
+
+func (s StoreHandler) GetRefValueIDs(
+	tenantID string,
+	trustAnchors []string,
+	claims map[string]interface{},
+) ([]string, error) {
+	platformClaimsMap, ok := claims["platform"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("claims do not contain platform map: %v", claims)
+	}
+	pids, err := arm.GetPlatformReferenceIDs(SchemeName, tenantID, platformClaimsMap)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get cca platform reference IDs: %w", err)
+	}
+	rids, err := realm.GetRealmReferenceIDs(SchemeName, tenantID, claims)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get cca realm reference IDs: %w", err)
+	}
+	return append(pids, rids...), nil
```

Comment:
Question: how can the caller tell a platform ID from a realm ID?

Reply:
Sorry, I did not get the question ? Perhaps clarify it better :)

Reply:
I mean the caller gets back:
```
[ a, b, c ]
```
How does it tell if "a" is a PID or RID?

Reply:
Thanks for clarifying the question. At this point in context the GetRefValueIDs() return an array of ReferenceValueIDs for the composite attester, to fetch the Reference Values as a collection. It does not need to know the distinction, which one is Platform and which one is Realm.

Reply:
OK, the caller may be just a conduit, but someone in the end must make sense of the list `[ "a", "b", "c" ]`, right?

Reply:
The flow is identical, whether the Attester is a Composite Attester or a Single Attester...

Reply:
Cool, thanks. That answers my question.